components/admin_console/text_setting.tsx

```diff
@@ -1,30 +1,36 @@
 // Copyright (c) 2015-present Mattermost, Inc. All Rights Reserved.
 // See LICENSE.txt for license information.
 
-import PropTypes from 'prop-types';
 import React from 'react';
 
 import TextSetting from 'components/widgets/settings/text_setting';
 
 import SetByEnv from './set_by_env';
 
-const AdminTextSetting = (props) => {
-    const {setByEnv, disabled, ...sharedProps} = props;
+interface Props {
+    id: string;
+    label: React.ReactNode;
+    value: string | number;
+    setByEnv: boolean;
+    disabled?: boolean;
+    sharedProps?: any[];
```

Comment:
sharedProps is not a prop, when you extract the props one by one using `const {setByEnv, id, label, value, ...sharedProps} = props;`, the `...sharedProps` at the ends means "the rest of the attributes of the object". That is used later to pass "the rest of the attributes of the object to the TextSetting component because the rest are completely equal.

Reply:
absolutely I thought it could be represented like that. I was wrong